src/Simplex/Messaging/Agent/Store.hs

```diff
@@ -1,31 +1,52 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DuplicateRecordFields #-}
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE KindSignatures #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE StandaloneDeriving #-}
 {-# OPTIONS_GHC -fno-warn-unticked-promoted-constructors #-}
 
-module Simplex.Messaging.Agent.Store
-  ( ReceiveQueue (..),
-    SendQueue (..),
-    Connection (..),
-    SConnType (..),
-    SomeConn (..),
-    MessageDelivery (..),
-    DeliveryStatus (..),
-    MonadAgentStore (..),
-  )
-where
+module Simplex.Messaging.Agent.Store where
 
+import Control.Exception (Exception)
+import Data.Int (Int64)
 import Data.Kind (Type)
-import Data.Time.Clock (UTCTime)
+import Data.Time (UTCTime)
 import Data.Type.Equality
-import Simplex.Messaging.Agent.Store.Types (ConnType (..))
 import Simplex.Messaging.Agent.Transmission
 import qualified Simplex.Messaging.Protocol as SMP
-import Simplex.Messaging.Types (RecipientPrivateKey, SenderPrivateKey, SenderPublicKey)
+import Simplex.Messaging.Types
+  ( MsgBody,
+    MsgId,
+    RecipientPrivateKey,
+    SenderPrivateKey,
+    SenderPublicKey,
+  )
+
+-- * Store management
+
+-- | Store class type. Defines store access methods for implementations.
+class Monad m => MonadAgentStore s m where
+  -- Queue and Connection management
+  createRcvConn :: s -> ReceiveQueue -> m ()
+  createSndConn :: s -> SendQueue -> m ()
+  getConn :: s -> ConnAlias -> m SomeConn
+  getRcvQueue :: s -> SMPServer -> SMP.RecipientId -> m ReceiveQueue
+  deleteConn :: s -> ConnAlias -> m ()
+  upgradeRcvConnToDuplex :: s -> ConnAlias -> SendQueue -> m ()
+  upgradeSndConnToDuplex :: s -> ConnAlias -> ReceiveQueue -> m ()
+  setRcvQueueStatus :: s -> ReceiveQueue -> QueueStatus -> m ()
+  setSndQueueStatus :: s -> SendQueue -> QueueStatus -> m ()
 
+  -- Msg management
+  createRcvMsg :: s -> ConnAlias -> MsgBody -> ExternalSndId -> ExternalSndTs -> BrokerId -> BrokerTs -> m ()
+  createSndMsg :: s -> ConnAlias -> MsgBody -> m ()
```

Comment:
Same here - there should be `SENT` acknowledgement sent to the client that contains internal ID